ctapipe/tools/tests/test_apply_models.py

```diff
@@ -0,0 +1,144 @@
+
+    assert f"{prefix}_tel_energy" in events.colnames
+    assert f"{prefix}_tel_is_valid" in events.colnames
+
+
+def test_apply_particle_classifier(
+    particle_classifier_path,
+    dl2_shower_geometry_file_lapalma,
+    tmp_path,
+):
+    from ctapipe.tools.apply_models import ApplyModels
+
+    input_path = dl2_shower_geometry_file_lapalma
+    output_path = tmp_path / "particle.dl2.h5"
+
+    ret = run_tool(
+        ApplyModels(),
+        argv=[
+            f"--input={input_path}",
+            f"--output={output_path}",
+            f"--particle-classifier={particle_classifier_path}",
+            "--StereoMeanCombiner.weights=konrad",
+        ],
+    )
+    assert ret == 0
+
+    table = read_table(
+        output_path, "/dl2/event/subarray/classification/ExtraTreesClassifier"
+    )
+    for col in "obs_id", "event_id":
+        assert table[col].description == EventIndexContainer.fields[col].description
+
+    for name, field in ParticleClassificationContainer.fields.items():
+        colname = f"ExtraTreesClassifier_{name}"
+        assert colname in table.colnames
+        assert table[colname].description == field.description
+
+    loader = TableLoader(output_path, load_dl2=True)
+    events = loader.read_subarray_events()
+    assert "ExtraTreesClassifier_prediction" in events.colnames
+    assert "ExtraTreesClassifier_telescopes" in events.colnames
+    assert "ExtraTreesClassifier_is_valid" in events.colnames
+    assert "ExtraTreesClassifier_goodness_of_fit" in events.colnames
+
+    events = loader.read_telescope_events()
+    assert "ExtraTreesClassifier_prediction" in events.colnames
+    assert "ExtraTreesClassifier_telescopes" in events.colnames
+    assert "ExtraTreesClassifier_is_valid" in events.colnames
+    assert "ExtraTreesClassifier_goodness_of_fit" in events.colnames
+
+    assert "ExtraTreesClassifier_tel_prediction" in events.colnames
+    assert "ExtraTreesClassifier_tel_is_valid" in events.colnames
+
+
+def test_apply_both(
+    energy_regressor_path,
+    particle_classifier_path,
+    dl2_shower_geometry_file_lapalma,
+    tmp_path,
+):
+    from ctapipe.tools.apply_models import ApplyModels
+
+    input_path = dl2_shower_geometry_file_lapalma
+    output_path = tmp_path / "particle-and-energy.dl2.h5"
+
+    ret = run_tool(
+        ApplyModels(),
+        argv=[
+            f"--input={input_path}",
+            f"--output={output_path}",
+            f"--particle-classifier={particle_classifier_path}",
+            f"--energy-regressor={energy_regressor_path}",
+            "--StereoMeanCombiner.weights=konrad",
+        ],
+    )
+    assert ret == 0
+
+    loader = TableLoader(output_path, load_dl2=True)
+
+    events = loader.read_subarray_events()
+    assert "ExtraTreesRegressor_energy" in events.colnames
+    assert "ExtraTreesClassifier_prediction" in events.colnames
+
+    events = loader.read_telescope_events()
+    assert "ExtraTreesClassifier_prediction" in events.colnames
```

Comment:
see above with `prefix`. Now I see the `prefix` and or configuration can be file-global